### ISSUE TITLE
npmrc ${ENVIRONMENT_VARIABLES} don't work as they should according to the npm docs

### ISSUE THREAD

**Issue by danielbayley** [pnpm/pnpm#696]
@zkochan Following on from #695…

> What do you mean? The default configs for local-registry, etc. are bad and you cannot override them?

Personally, it really annoys me when devs always just dump everything in the `~/.`/`$HOME` folder, so yeah I have different paths setup in my user-level `npmrc` (which lives in my _dotfiles_ repo, which I also _don't_ keep in `~`…) and this works fine with regular `npm` thanks to utilising `$NPM_CONFIG_USERCONFIG`.

Now, what I mean about `${ENVIRONMENT_VARIABLES}` not being expanded inside `npmrc` is explained here in the npm docs https://docs.npmjs.com/files/npmrc#files (or with more details https://docs.npmjs.com/misc/config#npmrc-files), for example you might set the line `init-author-name = ${GIT_AUTHOR_NAME}` in your `npmrc` rather than hard-coding your actual git author name, since you can set that environment variable, and use it elsewhere (like in git config) and this works fine with regular npm. But for some reason with _pnpm_, if `local-registry`/`store-path` or `global-path` values contain `${ENVIRONMENT_VARIABLES}` set in the same way, they are not expanded, and instead are read as literal `${STRINGS}`, resulting in actual `${FOLDER_PATHS_LIKE_THIS}` on disk! :/

What I am actually trying to utilise `${VARIABLES}` for is so that npm (and by extension _pnpm_) will respect the [_XDG spec_](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html), (which is a whole 'nother issue! But is no big deal if `${VARIABLES}` worked in `npmrc` as they should anyway…)

~~~ ini
# $DOTFILES/npm/npmrc
cache = ${XDG_CACHE_HOME}/npm # (XDG_CACHE_HOME is set to ~/Library/Caches…)

# pnpm
registry = https://registry.yarnpkg.com
local-registry = ${XDG_CACHE_HOME}/npm # Literally creates a `~/${XDG_CACHE_HOME}/npm` folder!
local-registry = ~/Library/Caches/npm # works though
store-path = ${XDG_CACHE_HOME}/npm/pnpm/store # fails
store-path = ~/Library/Caches/npm/pnpm/store # works
global-path = ${PREFIX}/lib #also fails (PREFIX is /usr/local…)
global-path = /usr/local/lib # works
~~~

**Comment by zkochan** [comment]
Seems like it is possible to use npm's API to load npm configs. I am not sure about the performance yet.

But it can be done like this

```js
const npm = require('npm')

npm.load()

npm.config.get('config-key')
//> config-value
```

**Comment by zkochan** [comment]
@danielbayley, I published the fix with [v0.65.0](https://github.com/pnpm/pnpm/releases/tag/v0.65.0). With the `next` dist-tag for now. Please test it when you'll have time
